[17.0][MIG] edi_oca: Migration to 17.0

john-herholz-dt opened the pull request:
A lot of views needed to be adjusted.

Furthermore I changed the _search method on edi_exchange_record, as the count param no longer works.
Not sure about the change as I don't know what the override of the _search method was suppossed to do.

Edit: @nguyenminhchien helped on the _search method.

etobella committed:
[IMP] edi: Security

simahawk committed:
edi: simply ack handling

ACK is now provided by a specific type of exchange
and relies on the relation between records of given type.

simahawk committed:
edi.backend: handle reception of files

simahawk committed:
edi: move and rename _exchange_notify_record to exc rec

Notify the related record belongs now to the exchange record itself
as it makes more sense. It has been renamed for better meaning
to '_notify_related_record'.

simahawk committed:
edi.backend: rename check method for exchange_process

simahawk committed:
edi: rename test file for process

simahawk committed:
edi: cron tests silent logger

simahawk committed:
edi: cron generate ack record when needed

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
edi 13.0.1.9.0

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
edi 13.0.1.10.0

simahawk committed:
edi: add receive component base mixin

simahawk committed:
edi: add type advanced settings

simahawk committed:
edi: fix manifest description

simahawk committed:
edi: add base lookup for validators

simahawk committed:
edi.backend: fix handle reception of files

simahawk committed:
edi.backend fix _check_input_exchange_sync signature

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
edi 13.0.1.11.0

simahawk committed:
edi: improve validation handling

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
edi 13.0.1.12.0

etobella committed:
[IMP] edi: Improve readme

etobella committed:
[IMP] edi: Allow to manually generate EDI Exchange records

OCA-git-bot committed:
edi 13.0.1.13.0

simahawk committed:
edi.backend: rename generate to match others

All the other main methods start with 'exchange_'.
Make generate respect this convention.

simahawk committed:
edi: add jobs configuration

etobella committed:
[FIX] edi: Rule not correctly created

OCA-git-bot committed:
edi 13.0.1.13.1

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
edi 13.0.1.14.0

etobella committed:
[IMP] edi: Add a has_exchange_record function to search for the specific exchange record related

etobella committed:
[IMP] edi: Keep old field values on the fields_view_get

Without this, we are loosing the specific views of some fields

OCA-git-bot committed:
edi 13.0.1.14.1

simahawk committed:
edi: consumer mixin fix Serialized field usage

simahawk committed:
edi: component ease access to record

simahawk committed:
edi: refactor component lookup

The match is now done based on specific control keys: backend type, exchange type.
This makes components more clean and their behavior more understandable.

OCA-git-bot committed:
edi 13.0.1.15.0

etobella committed:
[CHG] edi: Use more permissive licence: AGPL-> LGPL

simahawk committed:
edi: cron call exchange action w/ jobs

simahawk committed:
edi: enable cron for input records

simahawk committed:
edi: add TODO for cron methods

simahawk committed:
edi: _get_component fallback to backend model

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
edi 13.0.1.15.1

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
edi 13.0.1.16.0

acsonefho committed:
[FIX] edi: fix record property when there is no model to return the record's parent (if set) + unit test

etobella committed:
[FIX] edi: Fix security & filtering

etobella committed:
[FIX] edi_storage: sort key was not working properly

etobella committed:
[IMP] edi: Change security function

OCA-git-bot committed:
edi 13.0.1.17.0

etobella committed:
[IMP] edi_oca: black, isort, prettier

etobella committed:
[MIG] edi_oca: Migration to 14.0

oca-travis committed:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
[UPD] README.rst

etobella committed:
[FIX] edi_oca: Missing changes of name

OCA-git-bot committed:
edi_oca 14.0.1.0.1

sbidoul committed:
[IMP] update dotfiles

Yvesldff committed:
Added translation using Weblate (French)

Yvesldff committed:
Translated using Weblate (French)

Currently translated at 56.5% (78 of 138 strings)

Translation: edi-14.0/edi-14.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-edi_oca/fr/

LoisRForgeFlow committed:
[FIX] edi_oca: show the exchange file name while there is no file yet.

LoisRForgeFlow committed:
[FIX] edi_oca: show the exchange file name while there is no file yet.

etobella committed:
[FIX] edi: output might be not UTF-8 encodable, like PDFs

Co-authored-by: Núria Sancho <nuria.sancho@creublanca.es>

simahawk committed:
edi: add _find_existing_exchange_records to backend

LoisRForgeFlow committed:
[FIX] edi: allow the use of listeners.

use `_notify_*` methods when processing a exchange record. This
will allow the use of listeners based on this events.

oca-travis committed:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
edi_oca 14.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-14.0/edi-14.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-edi_oca/

OCA-git-bot committed:
edi_oca 14.0.1.2.0

simahawk committed:
edi_backend: update docstring

simahawk committed:
edi_backend: fix var name

simahawk committed:
edi: exc record get file content as b64 optionally

simahawk committed:
edi: improve exchange record UI

simahawk committed:
edi: improve backend view extension

simahawk committed:
edi: allow job channel conf by type

simahawk committed:
edi: specific method to get type settings

simahawk committed:
edi: load main vars at init

simahawk committed:
edi_backend: exclude class name from error txt

simahawk committed:
edi: exclude exception class from error msg

simahawk committed:
edi: improve exchange type and record UI

simahawk committed:
edi: add 'retry' action to exchange record

This action allows to re-schedule the record for the action that failed
depending on its current state and direction.

oca-travis committed:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
edi_oca 14.0.1.3.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-14.0/edi-14.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-edi_oca/

etobella committed:
[IMP] edi: Allow to ignore some jobs

oca-travis committed:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
edi_oca 14.0.1.4.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-14.0/edi-14.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-edi_oca/

simahawk committed:
edi: fix wrong status bar display new exc record

Without this check you get 2 overlapping bars till you select an exc type.

simahawk committed:
edi: fix consumer mixin

Exchange type has no display_name

etobella committed:
[IMP] edi: ext identifier unique by type and backend

It might be possible to have the same external_identifier for different backends and types.

OCA-git-bot committed:
edi_oca 14.0.1.5.0

OCA-git-bot committed:
edi_oca 14.0.1.5.1

simahawk committed:
edi: backend._get_component inject custom ctx

You can now provide additional env context from type settings.

oca-travis committed:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
edi_oca 14.0.1.6.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-14.0/edi-14.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-edi_oca/

OCA-git-bot committed:
edi_oca 14.0.1.6.1

simahawk committed:
edi_oca: remove filename from status msg

The filename is redundant there.
It's pointless to see it every time in the chatter.
If you want to see the details, you go to the exchange record.

simahawk committed:
edi: chatter mgs tmpl fix link

Linking via 'oe' attrs seems not to work.
Having a real URL also allows to open the record in a new tab.

simahawk committed:
edi: exchange record state use badge widget in tree view

This makes the state more prominent and clear.

etobella committed:
[IMP] edi_oca: Pass development status to Beta

OCA-git-bot committed:
[UPD] README.rst

i-vyshnevska committed:
[MIG][15.0] edi_oca

Commit:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
edi_oca 14.0.1.7.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-14.0/edi-14.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-edi_oca/

Commit:
[FIX] Fix access right issues when using actions

JordiMForgeFlow committed:
[FIX] edi_oca: fix chatter msg anchor

By adding the span element, the navigation is not performed correctly
and the anchor just redirects to blank '#'. Just remove the span.

Commit:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
edi_oca 15.0.1.0.1

simahawk committed:
edi: fix exc record name compute

No more ugly names in the form view till you populate all fields.

simahawk committed:
edi: improve exchanges views and menu

* add quick filters
* add quick menu items

simahawk committed:
edi: fix exc record missing _rec_name

simahawk committed:
edi: exc record form name not needed

simahawk committed:
edi: fix exc record sorting

simahawk committed:
edi: get rid of name, use identifier

Name contained redundant information like type name and related record.
On for and tree view is totally useless.
To still provide information on linked resource name
the 'related_name' field has been added.

simahawk committed:
edi: backend action to view exchanges

simahawk committed:
edi: backend action to view exchange types

simahawk committed:
edi: move job func to exc record

Prior to this change jobs were tied to the backend.

The backend must be responsible for processing records
and this is why the logic has been implemented there
and it's kept there with this change too.

However, it's way better if the job is tied to its record directly
because it makes more sense and it helps jumping to the record
w/ the 'related' btn on the job.

simahawk committed:
edi: update exc type adv settings help

Commit:
[UPD] Update edi_oca.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-15.0/edi-15.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-15-0/edi-15-0-edi_oca/

simahawk committed:
edi: show exchanges first

Makes little sense to show backends first
when you enter the EDI menu.
Ongoing exchanges are the most important thing.

simahawk committed:
edi_oca: fix exchange chatter link

simahawk committed:
edi_oca: avoid warning on exception args

etobella committed:
[FIX] edi_oca: Restore removed functions

We want to keep functionality on the same version. It is marked to remove for 16

Commit:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
edi_oca 15.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-15.0/edi-15.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-15-0/edi-15-0-edi_oca/

JordiMForgeFlow committed:
[IMP] edi_oca: add exchange_filename search

jcoux committed:
edi_oca: allow to override dt for exchange filename

By setting specific timezone and datetime format on exchange type
advanced settings, we can easily override the datetime used for generation of
exchange record filename.

OCA-git-bot committed:
edi_oca 15.0.1.2.0

Commit:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
edi_oca 15.0.1.3.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-15.0/edi-15.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-15-0/edi-15-0-edi_oca/

JaumeBforgeFlow committed:
[IMP] edi: archive edi_backend

simahawk committed:
edi: silent loggers in tests

simahawk committed:
edi_oca: add more generic mixin

Not all components might be bound to an exchange record.

simahawk committed:
edi_oca: add some TODOs

simahawk committed:
edi_oca: improve auto btns

* auto btns on/off flag per exchange type
* offer hook to customize how btns are rendered
* improve UI

simahawk committed:
edi_oca: update ownership

simahawk committed:
edi_oca: fix docstring on consumer mixin

simahawk committed:
edi: add meth _is_valid_edi_action

Handy method to validate core actions of the edi framework.

simahawk committed:
edi: improve consumer mixin helpers

simahawk committed:
edi: fix exc_rec.action_exchange_generate

Kewyord params were not propagated.

simahawk committed:
edi: improve exc_rec._trigger_edi_event

Propagate custom kw args to event handlers.

simahawk committed:
edi: fix process break on error ctx key

simahawk committed:
edi: improve create record validation

Raise specific error when exchange type is not found.

simahawk committed:
edi: ease creation of child records

simahawk committed:
edi: add edi_id_mixin

To be used to track references on exchanged related records.

simahawk committed:
edi_oca: bump 15.0.1.4.0

simahawk committed:
edi: fix consumer mixin test

simahawk committed:
edi: improve consumer mixin w/ origin

You can now track the original exchange record
that originated the document in the 1st place.

simahawk committed:
edi: add test for create child/ack

simahawk committed:
edi: add 'ack for' on type

Ease finding which type the current one is an ack for.

simahawk committed:
edi: improve exc type search

simahawk committed:
edi: fix ack record compute

You can produce or receive more than one ack for a given exchange.
If that's the case, make sure only the newest is considered.

simahawk committed:
edi: fix exchange ordering

In case exchange_on is the same, use ID to get the latest on top.

simahawk committed:
edi: fix record and type copy on fields

simahawk committed:
edi: fix error msg typo

simahawk committed:
edi: add exc.type.set_settings method

simahawk committed:
edi: improve chatter msg w/ type detail

simahawk committed:
edi: fix _cron_check_output_exchange_sync

Skip 'output_sent' records by default.
Reason: most of the times when a message is sent out
you don't care about its state anymore.
Any subsequent update from outside should come
with a new input record instead.

Yet, if for any reason this feature is required
you can still enable it by passing 'skip_sent=False'.

ivantodorovich committed:
[IMP] edi_oca: bump icon to 15.0 style

OCA-git-bot committed:
edi_oca 15.0.1.3.1

simahawk committed:
edi: allow search consumers by exc type

simahawk committed:
edi: fix action_view_edi_records

As we are reusing the existing window action
we must remove default filters
otherwise some related exchange records will be hidden by default.

simahawk committed:
edi: improve consumer record count perf

simahawk committed:
edi: fix record missing indexes

simahawk committed:
edi: add test for exc.record.action_retry

simahawk committed:
edi: fix exc.record._compute_retryable

simahawk committed:
edi: imp backend check input/output

Accept `record_ids` to work w/ specific records.

simahawk committed:
edi: improve cron names

simahawk committed:
edi: add quick exec option by type

Allow scheduling jobs immediately based on type configuration.
Very often you want an action to be executed right away.
You still get a job, meaning that the action is not sync
but the job will be scheduled immediately
instead of waiting for the cron to run.

simahawk committed:
edi: exchange type add TODO

simahawk committed:
edi: fix backward compat usage of _has_exchange_record_domain

In this commit https://github.com/OCA/edi/commit/0a9b6d5ef18f9d6c7d493ebefee8e579c697a165
I unified the way these kind of methods were used.

Was breaking older implementations where the method was called directly.

simahawk committed:
edi: drop ack record auto create

Creating the ack was kind of... an hack :)

When I added this behavior we didn't have yet the storage machinery
that looks automatically for a new file.
Also for webservices seems not relevant since we can configure endpoints
that can create the ack when needed.

All in all, I don't think this feature is needed.
Worse, it adds some overhead when a record gets created
and can generate write conflicts when exchanges are processed immediately
overlapping with the creation of the ack record.

simahawk committed:
edi: add 'as_bytes' option to _get_file_content

When sending the file data, we might want to use its bytes version.
This new flag gives an option for it.

simahawk committed:
edi: fix _get_component ctx propagation

The env_ctx that might come from a exchange type settings
was not propagated to the exchange record
which is made directly available to the work ctx of the component.

Before this change, if you had any method on the exchange record
or on the related record relying on a ctx key,
the ctx key was not propagated to all records' context.

victoralmau committed:
[FIX] edi_oca: Test errors

Co-authored-by: etobella

OCA-git-bot committed:
edi_oca 15.0.1.3.2

OCA-git-bot committed:
edi_oca 15.0.1.3.3

Commit:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
edi_oca 15.0.1.5.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-15.0/edi-15.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-15-0/edi-15-0-edi_oca/

john-herholz-dt committed:
[MIG] edi_oca: Migration to 16.0

florentx committed:
edi_oca: avoid ghost exchange records

florentx committed:
edi_oca: avoid backend mismatch

florentx committed:
edi_oca: assertion was not executed

florentx committed:
edi_oca: log warning for deleted records

simahawk committed:
edi: unify action complete notification

Now all main framework actions call the same notification method when complete.

simahawk committed:
edi: trigger generic event on action complete

simahawk committed:
edi: mark every session w/ 'edi_framework' ctx key

simahawk committed:
edi: trigger generic event on related record

simahawk committed:
edi: exc.record._set_related_record use sudo

Motivation: you end up using sudo on every call
because you don't know if the current user will have perm.
Yet, we don't care for such operatation.

simahawk committed:
edi: speed up exchange record views

* filter by default on the last 15 days w/ 'Recent' menu item
* filter by default on today on all menu items
* allow to browse them all w/ 'All' at the bottom

simahawk committed:
edi: add 'active' field to exchange type

simahawk committed:
edi: improve backend view filter inactive

simahawk committed:
edi: add disable flag to consumer

simahawk committed:
edi: allow enable/disable exc type by partner

simahawk committed:
edi: fix related record missing broken views

Before this change, when a related record was deleted
you could not access anymore any exchange record (tree or form)
due to the permission check.

Now:

* the check is bypassed
* related record buttons are hidden
* an informative message is displayed in the form

petrus-v committed:
edi_oca: fix consumer mixin exchange record count

simahawk committed:
edi: refactor model rules w/ specific model

Long awaited improvement.
Allows to decouple rules by model giving much more flexibility.
Migration steps included.

simahawk committed:
edi: unify menu item labels

simahawk committed:
edi: allow form btn actions customization

You can now customize and translate labels and tooltips for EDI automatic buttons.

simahawk committed:
edi: refactor model rules backward compat

simahawk committed:
edi_oca: fix test class names

Having the same name for different classes prevents selection of
specific test cases when running single test suites.

simahawk committed:
edi_oca: fix edi.exchange.type.create

simahawk committed:
edi_oca: move to OCA/edi-framework

simahawk committed:
edi_oca: use queue_job__no_delay key in tests

simahawk committed:
edi_oca: drop migrations

simahawk committed:
edi_oca: drop deprecated code

simahawk committed:
edi_oca: make not installable

simahawk committed:
edi: fix tests

simahawk committed:
edi: rename consumer mixin disable_edi_auto field

nilshamerlinck committed:
edi_oca: fix _logger.debug usage

TypeError: not all arguments converted during string formatting

simahawk committed:
edi: fix access errors tests

AccessError core messages were modified

Commit:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

simahawk committed:
edi: fix consumer get_view w/ nested forms

Select main  only as they can be nested into field custom forms.
I'm looking at you  on v16 :S

OCA-git-bot committed:
[BOT] post-merge updates

etobella committed:
[FIX] edi_oca: Show buttons properly. Use OWL

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update edi_oca.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-framework-16.0/edi-framework-16.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-framework-16-0/edi-framework-16-0-edi_oca/

OriolMForgeFlow committed:
[IMP] edi_oca: add company_id field with multi-company rule

MiquelRForgeFlow committed:
[FIX] edi_oca: avoid flooding error in the record chatter

Commit:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update edi_oca.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-framework-16.0/edi-framework-16.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-framework-16-0/edi-framework-16-0-edi_oca/

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[IMP] edi_oca: support exchange filename without extension on exchange record

OCA-git-bot committed:
[BOT] post-merge updates

john-herholz-dt commented on edi_oca/models/edi_exchange_record.py:
```diff
@@ -0,0 +1,572 @@
+        self._trigger_edi_event("done", suffix="ack_received_error")
+
+    @api.model
+    def _search(
```
Someone have a look at the changes here, please.

nguyenminhchien replied:
I have a suggestion here, please try it:

```

    @api.model
    def _search(self, domain, offset=0, limit=None, order=None, access_rights_uid=None):
        query = super()._search(
            domain,
            offset=offset,
            limit=limit,
            order=order,
            access_rights_uid=access_rights_uid,
        )

        if self.env.is_system():
            # restrictions do not apply to group "Settings"
            return query

        # TODO highlight orphaned EDI records in UI:
        #  - self.model + self.res_id are set
        #  - self.record returns empty recordset
        # Remark: self.record is @property, not field

        if query.is_empty():
            return query
        orig_ids = list(query)
        ids = set(orig_ids)
        result = []
        model_data = defaultdict(
            lambda: defaultdict(set)
        )  # {res_model: {res_id: set(ids)}}
        for sub_ids in self._cr.split_for_in_conditions(ids):
            self._cr.execute(
                """
                            SELECT id, res_id, model
                            FROM "%s"
                            WHERE id = ANY (%%(ids)s)"""
                % self._table,
                dict(ids=list(sub_ids)),
            )
            for eid, res_id, model in self._cr.fetchall():
                if not model:
                    result.append(eid)
                    continue
                model_data[model][res_id].add(eid)

        for model, targets in model_data.items():
            if not self.env[model].check_access_rights("read", False):
                continue
            recs = self.env[model].browse(list(targets))
            missing = recs - recs.exists()
            if missing:
                for res_id in missing.ids:
                    _logger.warning(
                        "Deleted record %s,%s is referenced by edi.exchange.record %s",
                        model,
                        res_id,
                        list(targets[res_id]),
                    )
                recs = recs - missing
            allowed = (
                self.env[model]
                .with_context(active_test=False)
                ._search([("id", "in", recs.ids)])
            )
            for target_id in allowed:
                result += list(targets[target_id])
        if len(orig_ids) == limit and len(result) < len(orig_ids):
            extend_query = self._search(
                domain,
                offset=offset + len(orig_ids),
                limit=limit,
                order=order,
                access_rights_uid=access_rights_uid,
            )
            extend_ids = list(extend_query)
            result.extend(extend_ids[: limit - len(result)])
        # Restore original ordering
        result = [x for x in orig_ids if x in result]
        if set(orig_ids) != set(result):
            # Create a virgin query
            query = self.browse(result)._as_query()
        return query
```

john-herholz-dt replied:
Thanks for the suggestion!

nguyenminhchien commented on edi_oca/__manifest__.py:
```diff
@@ -0,0 +1,51 @@
+    Define backends, exchange types, exchange records,
+    basic automation and views for handling EDI exchanges.
+    """,
+    "version": "17.0.1.1.0",
```
it should be `17.0.1.0.0`

nguyenminhchien commented on edi_oca/migrations/16.0.1.0.0/pre-migrate.py:
please remove the `migrations` folder

nguyenminhchien commented on edi_oca/models/edi_exchange_record.py:
```diff
@@ -0,0 +1,572 @@
+            return res.decode() if not as_bytes else res
+        return self[field_name]
+
+    def name_get(self):
```
you should now override `_compute_display_name` instead, adding the possible new fields in the depends of the method. See https://github.com/odoo/odoo/pull/122085 for more details.
For reference:
```
    @api.depends("identifier", "res_id", "model", "type_id")
    def _compute_display_name(self):
        for rec in self:
            rec_name = rec.identifier
            if rec.res_id and rec.model:
                rec_name = rec.record.display_name
            rec.display_name = f"[{rec.type_id.name}] {rec_name}"

```
Seems that you miss this document: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-17.0#tasks-to-do-in-the-migration 
please follow its guideline

nguyenminhchien commented on edi_oca/static/src/xml/widget_edi.xml:
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<templates id="template" xml:space="preserve">
+    <div t-name="edi_oca.EdiConfigurationWidget" owl="1">
```
please remove `owl="1"`

nguyenminhchien commented on edi_oca/views/edi_exchange_type_rule_views.xml:
```diff
@@ -0,0 +1,124 @@
+        <field name="model">edi.exchange.type.rule</field>
+        <field name="arch" type="xml">
+            <tree decoration-muted="(not active)">
+                <field name="active" invisible="1" />
```
```suggestion
                <field name="active" column_invisible="1" />
```

nguyenminhchien commented on edi_oca/views/edi_exchange_type_views.xml:
```diff
@@ -0,0 +1,158 @@
+                                context="{'default_type_id': active_id}"
+                            >
+                                <tree decoration-muted="(not active)">
+                                    <field name="active" invisible="1" />
```
```suggestion
                                    <field name="active" column_invisible="1" />
```

nguyenminhchien reviewed (changes requested):
PR of `base_edi` has been merged, you can remove it from `test-requirements.txt`

john-herholz-dt commented:
Thanks @nguyenminhchien for your in depth review. Will adjust everything later today.
Sure I might have forgotten about some essential changes like the name_get. I didn't look into it.

john-herholz-dt commented:
I have applied the suggested changes, all tests now pass.

Commit:
[IMP] edi_oca: add partner form page

Commit:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-framework-16.0/edi-framework-16.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-framework-16-0/edi-framework-16-0-edi_oca/

nilshamerlinck committed:
[FIX] edi_oca: tests using FakeModelLoader should not be post_install

OCA-git-bot committed:
[BOT] post-merge updates

john-herholz-dt commented:
Hello @simahawk , I know you are busy, but once you have some minutes, could you check this PR, I need to migrate everything to 17.0 and edi_oca is the most critical.

I also saw there was another PR for 16 "add partner form page". Do I need to update the code?
Does this work with the git patch command from the migration wiki?

Thank you very much!

john-herholz-dt committed:
[IMP] edi_oca: pre-commit auto fixes

john-herholz-dt committed:
[MIG] edi_oca: Migration to 17.0

john-herholz-dt committed:
[DON'T MERGE] test-requirements

john-herholz-dt commented:
I have re-done the migration with the latest changes from 16.0 branch that were not included previously.

john-herholz-dt commented:
@simahawk what are we doing with this?

JordiMForgeFlow reviewed (approved):
Code review

peluko00 commented:
Can you rebase please for the runboat @john-herholz-dt ?

john-herholz-dt commented:
I did the rebase, somehow it had now problems with the pre-commit checks, I did another one pre-commit on my side

peluko00 commented:
the last commit will be include with the migration commit

> I did the rebase, somehow it had now problems with the pre-commit checks, I did another one pre-commit on my side

peluko00 reviewed (approved):
Code review, LGTM

gurneyalex commented:
/ocabot merge nobump

john-herholz-dt commented:
Finally! Jeehaaaa.
